circus-train-s3-mapreduce-cp-copier/src/main/java/com/hotels/bdp/circustrain/s3mapreducecpcopier/S3MapReduceCpCopier.java

```diff
@@ -57,6 +57,26 @@ public Job exec(Configuration conf, S3MapReduceCpOptions options) throws Excepti
 
   private static final Logger LOG = LoggerFactory.getLogger(S3MapReduceCpCopier.class);
 
+  private static class JobCounter implements Gauge<Long> {
```

Comment:
I think this could be factored into a more generic `JobCounterGauge` which has an additional constructor which takes the group name. If this is set it uses it in the call to `findCounter`, if not it does the below. Then you can replace the above `FileSystemCounterGauge` with it too.